Intermittent failure: ServiceProxyTest.testConnectionTimeout

```
Pile d'exécution

java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at io.vertx.test.core.AsyncTestBase.assertTrue(AsyncTestBase.java:372)
	at io.vertx.serviceproxy.test.ServiceProxyTest.lambda$null$95(ServiceProxyTest.java:860)
Sortie d'erreur standard

Starting test: ServiceProxyTest#testConnectionTimeout 
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at io.vertx.test.core.AsyncTestBase.assertTrue(AsyncTestBase.java:372)
	at io.vertx.serviceproxy.test.ServiceProxyTest.lambda$null$95(ServiceProxyTest.java:860)
	at io.vertx.core.eventbus.impl.HandlerRegistration.deliver(HandlerRegistration.java:212)
	at io.vertx.core.eventbus.impl.HandlerRegistration.handle(HandlerRegistration.java:191)
	at io.vertx.core.eventbus.impl.EventBusImpl.lambda$deliverToHandler$3(EventBusImpl.java:524)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:337)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:445)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:745)
Unhandled exception 
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at io.vertx.test.core.AsyncTestBase.assertTrue(AsyncTestBase.java:372)
	at io.vertx.serviceproxy.test.ServiceProxyTest.lambda$null$95(ServiceProxyTest.java:860)
	at io.vertx.core.eventbus.impl.HandlerRegistration.deliver(HandlerRegistration.java:212)
	at io.vertx.core.eventbus.impl.HandlerRegistration.handle(HandlerRegistration.java:191)
	at io.vertx.core.eventbus.impl.EventBusImpl.lambda$deliverToHandler$3(EventBusImpl.java:524)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:337)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:445)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:745)
```

https://vertx.ci.cloudbees.com/job/vert.x3-service-proxy/2720
https://vertx.ci.cloudbees.com/job/vert.x3-service-proxy/2711
https://vertx.ci.cloudbees.com/job/vert.x3-service-proxy/2709

--- a/src/test/java/io/vertx/serviceproxy/test/ServiceProxyTest.java
+++ b/src/test/java/io/vertx/serviceproxy/test/ServiceProxyTest.java
@@ -39,6 +39,8 @@ import java.util.Set;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
+import static java.util.concurrent.TimeUnit.*;
+
 /**
  * @author <a href="http://tfox.org">Tim Fox</a>
  */
@@ -848,16 +850,16 @@ public class ServiceProxyTest extends VertxTestBase {
     consumer = ProxyHelper.registerService(TestService.class, vertx, service, SERVICE_ADDRESS, timeoutSeconds);
 
     proxy.createConnection("foo", onSuccess(conn -> {
+      long start = System.nanoTime();
+
       conn.startTransaction(onSuccess(res -> {
         assertEquals("foo", res);
-
-        long start = System.currentTimeMillis();
 
         vertx.eventBus().consumer("closeCalled").handler(msg -> {
           assertEquals("blah", msg.body());
 
-          long now = System.currentTimeMillis();
-          assertTrue(now - start > timeoutSeconds * 1000);
+          long duration = System.nanoTime() - start;
+          assertTrue(String.valueOf(duration), duration >= SECONDS.toNanos(timeoutSeconds));
 
           // Should be closed now
           conn.startTransaction(onFailure(cause -> {

